modules/ivc_champva/app/models/ivc_champva/ves_request.rb

```
@@ -0,0 +1,166 @@
+# frozen_string_literal: true
+
+module IvcChampva
+  class VesRequest
+    attr_accessor :application_type, :application_uuid, :sponsor, :beneficiaries, :certification
+
+    def initialize(params = {})
+      @application_type = params[:application_type] || 'CHAMPVA'
+      @application_uuid = params[:application_uuid] || SecureRandom.uuid
+      @sponsor = Sponsor.new(params[:sponsor] || {})
+      @beneficiaries = (params[:beneficiaries] || []).map { |ben| Beneficiary.new(ben) }
+      @certification = Certification.new(params[:certification] || {})
+    end
+
+    def to_json(*_args)
+      {
+        applicationType: @application_type,
+        applicationUUID: @application_uuid,
+        sponsor: @sponsor.to_hash,
+        beneficiaries: @beneficiaries.map(&:to_hash),
+        certification: @certification.to_hash
+      }
+    end
+
+    class Sponsor
+      attr_accessor :first_name, :last_name, :middle_initial, :suffix, :ssn, :va_file_number,
+                    :date_of_birth, :date_of_marriage, :date_of_death, :is_deceased,
+                    :is_death_on_active_service, :phone_number, :address, :person_uuid
+
+      def initialize(params = {})
+        @person_uuid = params[:person_uuid] || params[:personUUID]
```

Should we allow the keynames to vary in casing like this? Or should we just enforce `snake_case` and let it fail otherwise?

---

Yes you're right - addressing this now.